Add Reset method to delete key from Scratch

paulcmal opened the pull request:
Following a [discussion](https://discourse.gohugo.io/t/delimit-and-create-array/11065/11?u=cmal) in the forums, I'm proposing a `Reset` method for Scratch, to remove a Scratch entry. It could be renamed Delete if you prefer.

This is my first time writing some golang so sorry if I'm doing some things wrong. Unfortunately, I'm running golang 1.8.1 from Debian stable so I cannot run the tests. I tried it with the following snippet and it appears to work like a charm:

```
{{ $.Scratch.Set "test" "lol" }}
{{ $.Scratch.Reset "test" }}
{{ $.Scratch.Reset "anothertest" }}
{{ $.Scratch.Add "test" "Bonjour" }}
{{ $.Scratch.Get "test" }}
```

It printed "Bonjour" like expected and did not raise an error when attempting to delete a key that did not exist (I believe this is the desired behaviour).

I read the contribution guidelines and tried to give meaning to my commit message. Don't hesitate to amend my commit :)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gohugoio/hugo?pullRequest=4509) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gohugoio/hugo?pullRequest=4509) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gohugoio/hugo?pullRequest=4509) it.</sub>

bep commented:
Great, we can add this if you:

1. Rename it to "Delete". This is more in line with what it does.
2. Sign the CLA above (if it does not work, check your commit email address vs the addresses you have registered in GitHub)
3. Update the Scratch docs. Do it in this repository.

paulcmal commented:
Thank you for the quick answer @bep!

I added an example in the docs, renamed `Reset` to `Delete`. Unfortunately, I can't figure out why the tests are failing, since I can't run them locally.

I tried to upgrade to golang 1.10 manually. That worked (`go version go1.10 linux/amd64`) but even after cleaning/redownloading/recompiling everything, I still get `Skip Check on go1.8.1`. When I compile my own small hello world, I get runtime version 1.10 as expected. Do you have any idea what could be causing this?

bep commented:
You need to run `gofmt` on your code.

bep commented:
@paulcmal give LiteIDE a try. It does gofmt on save -- it is really fast and easy to use and my fav editor for Go code. And it is free ...
